src/renderer/components/settings-execution.tsx

```diff
@@ -12,13 +12,20 @@ import {
 } from '@blueprintjs/core';
 import { observer } from 'mobx-react';
 
-import { IPackageManager } from '../../interfaces';
+import {
+  GlobalSetting,
+  GlobalSettingKey,
+  IPackageManager,
+} from '../../interfaces';
 import { AppState } from '../state';
 
-export enum SettingItemType {
-  EnvVars = 'environmentVariables',
-  Flags = 'executionFlags',
-}
+/**
+ * @TODO make this a proper enum again once we update Typescript
+ */
+export const SettingItemType = {
+  EnvVars: GlobalSetting.environmentVariables,
+  Flags: GlobalSetting.executionFlags,
+} as const;
```

Comment:
Could you provide further context here on what the issue is, and which version of TypeScript will fix it?

Reply:
Sure. The problem is I can't compute the values of the `SettingItemType` enum using another enum. We used to have the enum values as string literals...
```ts
export enum SettingItemType {
  EnvVars = 'environmentVariables',
  Flags = 'executionFlags',
}
```
...but since these values are keys used in the localStorage, I moved them to the `GlobalSetting` enum, so I'm trying to keep these values in sync by using the value from `GlobalSetting` in `SettingItemType`...

```ts
enum GlobalSetting {
  // ...
  environmentVariables = 'environmentVariables',
  executionFlags = 'executionFlags',
  // ...
}

export enum SettingItemType {
  EnvVars = GlobalSetting.environmentVariables,
  Flags = GlobalSetting.executionFlags,
}
```
...but Typescript won't let me do this using `SettingItemType` as an enum:

![image](https://github.com/electron/fiddle/assets/19478575/ba09b493-502e-4a40-8eff-96e7c53d0ba2)

I've changed the signature of some methods that deal with the localStorage to accept a `GlobalSettingKey` (the union of all possible keys in `GlobalSetting`) instead of a plain `string`, and Typescript gets mad at me. However, it works fine if I use an object instead of an enum, so I did that.

Apparently, this is one of the "long-standing oddities around enums" mentioned in the [5.0 release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#enum-overhaul), so changing `SettingItemType` back to an enum would require TS 5.0.

We could also just get rid of `SettingItemType` and refactor any usages to grab the values from `GlobalSetting` directly, but since that enum was already there, I didn't want to do that just yet to keep the changes as directed as possible.